scripts/svdpatch.py

```diff
@@ -486,6 +543,10 @@ def process_peripheral(svd, pspec, peripheral, update_fields=True):
     pcount = 0
     for ptag in iter_peripherals(svd, pspec):
         pcount += 1
+        # Handle register arrays
+        for rspec in peripheral.get("_array", {}):
+            rmod = peripheral["_array"][rspec]
+            process_peripheral_regs_array(ptag, rspec, rmod)
```

Comment:
Please could you move array handling to below addition handling? We might need to add/modify/delete registers that will become part of the array, which is impossible if array handling is processed first.